### ISSUE TITLE
RFC: Allow for selectively skipping local resolver with a hook option

### ISSUE THREAD

**Issue by mtourj** [urql-graphql/urql#2650]
## Summary

Graphcache is great and the local resolvers feature works great for merging data together when doing cursor pagination. However, if I wanted to use the same query in multiple places and in some components I did NOT want to merge data when paginating, my hands are tied because if you apply a local resolver to a query, then it applies to ALL calls to that query. I tried setting `requestPolicy` to `network-only` but that does not seem to have an effect on whether local resolvers are used.

## Proposed Solution

Perhaps a simple switch to skip the local resolver straight from the hooks. Maybe a skipLocalResolvers option or something similar? 

## Requirements

I am not familiar with the internals of urql, so really I'm just looking to have an option to decide whether a query will use the local resolvers I've written to resolve things or not, however that might best be implemented.

Thank you for the amazing work! Love using urql  :)

**Comment by focux** [comment]
This would be really useful, I'm having the same experience as you.

**Comment by JoviDeCroock** [comment]
Generally I would say this option already exists, let's say for a local resolver you are using the `pagination` but you want to opt out of it then you can do

```
const myResolver = relayPagination();

cacheExchange({
  resolvers: {
    Query: {
      list: (parent, args, cache, info) => {
        // or any other condition 😅 
        if (info.variables.skipPagination) {
          return cache.resolve('Query', 'list', args)
        }

        return myResolver(parent, args, cache, info);
      }
    }
  }
})
```

**Comment by mtourj** [comment]
Thanks @JoviDeCroock, that's very helpful.

**Comment by mtourj** [comment]
> Generally I would say this option already exists, let's say for a local resolver you are using the `pagination` but you want to opt out of it then you can do
> 
> ```js
> const myResolver = relayPagination();
> 
> cacheExchange({
>   resolvers: {
>     Query: {
>       list: (parent, args, cache, info) => {
>         // or any other condition 😅 
>         if (info.variables.skipPagination) {
>           return cache.resolve('Query', 'list', args)
>         }
> 
>         return myResolver(parent, args, cache, info);
>       }
>     }
>   }
> })
> ```

Just tried this - `info.variables` seems to be getting filtered down before it is passed to cacheExchange resolvers. @JoviDeCroock , so I cannot do conditional behavior this way

**Comment by JoviDeCroock** [comment]
It shouldn't be based on the code [here](https://github.com/urql-graphql/urql/blob/af93ef9ab9595478d8489a6280b08a731b1d7faa/exchanges/graphcache/src/ast/variables.ts#L49-L72) can look at it next week though

**Comment by mtourj** [comment]
Sounds good, much appreciated.

Yeah I'm not sure where it happens but it's definitely being passed in to useQuery and not making it into the `info` object.

For context I'm on Graphcache 4.4.3, although it seems that neither of the snippets you linked have changed since then, so might not be a relevant detail.

**Comment by JoviDeCroock** [comment]
Just tried in a sanbox, works great https://codesandbox.io/s/jovial-cloud-usm1t4?file=/src/components/Todos.js

**Comment by mtourj** [comment]
Okay I got it working, it turned out to be filtering it out because I had `requestPolicy: "network-only"` set, thank you :)

**Comment by kitten** [comment]
Resolved by #3191 / #3306 which have been released in https://github.com/urql-graphql/urql/releases/tag/%40urql%2Fexchange-graphcache%406.2.0
